site/blog/2016/jul/1/sphinx-and-rtd-for-writers.rst

```diff
@@ -0,0 +1,548 @@
+
+     install
+     support
+
+The above example outputs a Table of Contents in the page where it occurs.
+The ``maxdepth`` argument tells Sphinx to include 2 levels of headers.
+This also tells Sphinx that the other pages are sub-pages of the current page,
+creating a "tree" structure of the pages::
+
+    index
+    ├── install
+    ├── support
+
+The TOC Tree is also used for generating the global navigation inside of Sphinx.
+The `index` at the top of your project acts as the root of the navigation.
+The ``toctree`` is quite important,
+and one of the most powerful concepts in Sphinx.
+
+More info on the ``toctree`` can be found here:
+
+* http://www.sphinx-doc.org/en/stable/markup/toctree.html
+
+Working with Code
+`````````````````
+
+Showing code examples is a common task in all documentation.
+Sphinx is quite well prepared for this task.
+Let's see how we can show a basic code example::
+
+    .. code-block:: python
+       :linenos:
+
+       import antigravity
+
+       def main():
+           antigravity.fly()
+
+This would display the Python snippet,
+syntax highlighted,
+with line numbers.
+The ``python`` in the above example is an argument to the ``code-block`` *directive*.
+The ``:linesnos:`` is an *option* that says to display line numbers for the code block.
+In the HTML output,
+it would look like this:
+
+.. image:: code.png
+
+Sphinx doesn't stop there though.
+It allows you to store your code examples in external files,
+and be included in your documentation for easier maintenance.
+This uses the ``literalinclude`` directive::
+
+    .. literalinclude:: example.rb
+       :language: ruby
+       :emphasize-lines: 12,15-18
+
+The neat thing here is the ``:emphasize-lines:``.
+This shows the lines highlighted in your output.
+This is quite useful for showing sets of code examples where a subset of the code changes.
+You can also specify just a subset of lines to show with the ``:lines:`` option,
+so you don't have to manage multiple snippets.
+
+These directives have a lot more powerful options than I can show in this article.
+You can see the full documentation for them on the Sphinx website:
+
+* http://www.sphinx-doc.org/en/stable/markup/code.html
+
+Working with References
+```````````````````````
+
+A powerful reference system is a large part of the power of Sphinx.
+You are able to reference arbitrary headings and paragraphs in your content,
+but also semantically reference a large number of things as well.
+On top of that,
+Sphinx includes ``intersphinx`` which allows referencing across Sphinx projects.
+This means that if you have multiple projects inside your company,
+you can still use semantic referencing across them!
+
+A simple reference is defined as such::
+
+    .. _reference-target-name::
+
+    This is a bit of content.
+
+    This is how you point to the above reference, :ref:`reference-target-name`
+
+Sphinx also includes a number of other semantic reference types.
+Here are a few examples:
```

Comment:
"Examples of other semantic reference types that Sphinx provides:" (yes, fragment is OK here)